src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/PrivilegesInterceptorImpl.java

```diff
@@ -203,6 +224,7 @@ private void replaceIndex(final ActionRequest request, final String oldIndexName
                 }
 
                 if (ar instanceof IndexRequest) {
+                    createIndexRequest = checkAndCreateRequestIfAbsent(newIndexName);
```

Comment:
Not sure how often there might be bulk requests, but do you think that it's worth to add some optimization here, and generate the object only once for the first Index Request (in case if there are N write requests to the same kibana+tenant index).